app/src/main/java/unipd/se18/ocrcamera/AllergensManager.java

```diff
@@ -0,0 +1,254 @@
+package unipd.se18.ocrcamera;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import com.opencsv.CSVReader;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+
+
+
+/**
+ * Class that contains the methods to read the allergens from a file, create an allergen list,
+ * find the allergens in the recognized ingredients from the OCR.
+ */
+class AllergensManager {
+
+    private final int COMMON_NAME = 0;
+    private final int INCI_NAMES = 1;
+
+    private ArrayList<Allergen> allergensList = new ArrayList<>();
+
+    private Context context;
+    private SharedPreferences sp = null;
```

Comment:
I don't understand why this variable is actually initialized as null, considering the fact that this particular value isn't used after that if not to be overwritten inside the constructor. I think in this case it would be better not to initialize it at all.